app/src/main/java/org/simple/clinic/patient/ReminderConsent.kt

```diff
@@ -0,0 +1,55 @@
+package org.simple.clinic.patient
+
+import android.os.Parcelable
+import androidx.annotation.VisibleForTesting
+import androidx.room.TypeConverter
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+import kotlinx.android.parcel.Parcelize
+import org.simple.clinic.util.SafeEnumTypeAdapter
+
+sealed class ReminderConsent : Parcelable {
+
+  @Parcelize
+  object Granted : ReminderConsent()
+
+  @Parcelize
+  object Denied : ReminderConsent()
+
+  @Parcelize
+  data class Unknown(val actualValue: String) : ReminderConsent()
+
+  object TypeAdapter : SafeEnumTypeAdapter<ReminderConsent>(
+      knownMappings = mapOf(
+          Granted to "granted",
+          Denied to "denied"
+      ),
+      unknownStringToEnumConverter = { Unknown(it) },
+      unknownEnumToStringConverter = { (it as Unknown).actualValue }
+  )
+
+  class RoomTypeConverter {
+
+    @TypeConverter
+    fun toEnum(value: String?): ReminderConsent? = TypeAdapter.toEnum(value)
+
+    @TypeConverter
+    fun fromEnum(status: ReminderConsent?): String? = TypeAdapter.fromEnum(status)
+  }
+
+  class MoshiTypeAdapter {
+
+    @FromJson
+    fun fromJson(value: String?): ReminderConsent? = TypeAdapter.toEnum(value)
+
+    @ToJson
+    fun toJson(status: ReminderConsent?): String? = TypeAdapter.fromEnum(status)
+  }
+
+  companion object {
+
+    @VisibleForTesting
+    fun random() = TypeAdapter.knownMappings.keys.shuffled().first()
+  }
```

Comment:
The `random()` function isn't being used in tests since we have hard-coded it to `Granted`. I think we can remove this.